PowerShellSyntax.tmLanguage

```diff
@@ -13,1598 +13,7352 @@
+											<string>(?=.|$)</string>
+											<key>applyEndPatternLast</key>
+											<true/>
+											<key>patterns</key>
+											<array>
+												<dict>
+													<key>comment</key>
+													<string>allow for possibility that numeric value or unary operators match first, before resuming in expression mode</string>
+													<key>begin</key>
+													<string>(?!\G)(?![\n;&amp;|)}\]])</string>
+													<key>end</key>
+													<string>(?=[\n;)}\]])</string>
+													<key>patterns</key>
+													<array>
+														<dict>
+															<key>include</key>
+															<string>#expression_mode</string>
+														</dict>
+													</array>
+												</dict>
+												<dict>
+													<key>include</key>
+													<string>#numericConstant</string>
+												</dict>
+												<dict>
+													<key>include</key>
+													<string>#operators_preUnary</string>
+												</dict>
+												<dict>
+													<key>comment</key>
+													<string>if neither numeric or a pre-unary operator, just finish in expression mode</string>
+													<key>begin</key>
+													<string>(?![\n;&amp;|)}\]])</string>
+													<key>end</key>
+													<string>(?=[\n;)}\]])</string>
+													<key>patterns</key>
+													<array>
+														<dict>
+															<key>include</key>
+															<string>#expression_mode</string>
+														</dict>
+													</array>
+												</dict>
+											</array>
+										</dict>
+										<dict>
+											<key>include</key>
+											<string>#commentBlock</string>
+										</dict>
+										<dict>
+											<key>comment</key>
+											<string>normally provided by command mode, but thats not available here</string>
+											<key>match</key>
+											<string>`\n</string>
+											<key>name</key>
+											<string>constant.character.escape.powershell</string>
+										</dict>
+									</array>
+								</dict>
+								<dict>
+									<key>include</key>
+									<string>#notCode</string>
+								</dict>
+							</array>
+						</dict>
+						<dict>
+							<key>comment</key>
+							<string>`,`, `&amp;` and `|` not permitted here</string>
+							<key>match</key>
+							<string>[,&amp;|]</string>
+							<key>name</key>
+							<string>invalid.source.powershell</string>
+						</dict>
+						<dict>
+							<key>include</key>
+							<string>#notCode</string>
+						</dict>
+					</array>
+				</dict>
+				<dict>
+					<key>include</key>
+					<string>#notCode</string>
+				</dict>
+			</array>
 		</dict>
 		<dict>
 			<key>begin</key>
-			<string>(?&lt;!\w)(--%)(?!\w)</string>
+			<string>(?i:function|filter|workflow)(?=[\s{(,;&amp;|)}])</string>
 			<key>beginCaptures</key>
 			<dict>
-				<key>1</key>
+				<key>0</key>
 				<dict>
 					<key>name</key>
-					<string>keyword.control.powershell</string>
+					<string>keyword.other.$0.powershell</string>
```

Comment:
This should be storage.type.function.powershell and keyword.declaration.function.powershell in addition to keyword.other to be in line with the sublime text scope naming recommendations as well
https://www.sublimetext.com/docs/3/scope_naming.html